modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/types.ts

```diff
@@ -132,6 +132,32 @@ export interface ILiferay {
 	readonly currentURLEncoded: string;
 }
 
+// Properties from `frontend-js-web/events`
+
+export type Callback = (...args: any[]) => void;
+export interface ILiferay {
+	after(eventName: string, callback: Callback): void;
+	before(eventName: string, callback: Callback): void;
+	on(eventName: string, callback: Callback): void;
+	once(eventName: string, callback: Callback): void;
+	onceAfter(eventName: string, callback: Callback): void;
+}
+
+// Properties from `frontend-js-spa-web/init`
+
+export interface ILiferay {
```

Comment:
Will the second `export interface ILiferay` override the first?

Reply:
No, they merge as per [Declaration Merging](https://www.typescriptlang.org/docs/handbook/declaration-merging.html) which is what I thought we could leverage to compose them from different modules

Reply:
Interesting! Looks powerful, but also quite quirky, with a lot of edge cases.